dist/tools/compile_and_test_for_board/compile_and_test_for_board.py

```diff
@@ -101,15 +103,41 @@ def apps_directories(riotdir, apps_dirs=None, apps_dirs_skip=None):
     :param apps_dirs_skip: list of application directories to remove, applied
                            on the RIOT list or `apps_dirs`
     """
-    apps_dirs = apps_dirs or _riot_tracked_applications_dirs(riotdir)
-    apps_dirs_skip = apps_dirs_skip or []
+    apps_dirs = _matching_apps_directories(apps_dirs, riotdir)
+    apps_dirs_skip = _matching_apps_directories(apps_dirs_skip, riotdir,
+                                                exclude=True)
 
     # Remove applications to skip
     apps_dirs = set(apps_dirs) - set(apps_dirs_skip)
 
     return sorted(list(apps_dirs))
 
 
+def _matching_apps_directories(apps_dirs, riotdir, exclude=False):
```

Comment:
I would not put all this in a sub-function. Just convert `apps_dirs` to `apps_dirs with wildcard resolution`
This function is used with two modes which increases complexity

Reply:
It completely changes the behavior of the previous implementation.

nothing == RIOT directories
applications == use these ones, not considering whit RIOT is giving you so can force non tracked apps.
Exclude: exclusion that are applied after to blacklist only one test.